crates/solver/src/settlement_submission.rs

```diff
@@ -148,44 +151,51 @@ impl SolutionSubmitter {
         account: Account,
         nonce: U256,
     ) -> Result<TransactionReceipt, SubmissionError> {
-        let is_dry_run: bool = self
-            .transaction_strategies
-            .iter()
-            .any(|strategy| matches!(strategy, TransactionStrategy::DryRun));
+        for strategy in &self.transaction_strategies {
+            match strategy {
+                TransactionStrategy::DryRun => {
+                    return Ok(dry_run::log_settlement(account, &self.contract, settlement).await?);
+                }
+                TransactionStrategy::Gelato(gelato) => {
+                    return tokio::time::timeout(
+                        self.max_confirm_time,
+                        gelato.relay_settlement(account, settlement),
+                    )
+                    .await
+                    .map_err(|_| SubmissionError::Timeout)?;
+                }
```

Comment:
If we have `gelato` in our list of transactions strategies, than no other strategies will even be started? (if I understand the code correctly)

Reply:
@sunce86 - added a clarifying comment.

Reply:
Understood. I think this is fine for now. Thanks for adding a comment.